src/java/org/jivesoftware/openfire/muc/spi/LocalMUCRoom.java

```diff
@@ -1439,6 +1439,8 @@ private Presence changeOccupantRole(JID jid, MUCRole.Role newRole) throws NotAll
             if (role.isLocal()) {
                 // Update the presence with the new role
                 role.setRole(newRole);
+                // Set the new presence, so that the updated role is reflected in the presence stanza.
+                role.setPresence(role.getPresence());

```

Comment:
I see this works because setPresence() rebuilds LocalMUCRole.presence with the new extendedInformation - is it worth causing this to happen within setRole()/setAffiliation() instead?